lib/ansible/plugins/action/include_vars.py

```diff
@@ -128,6 +129,11 @@ def run(self, tmp=None, task_vars=None):
             scope[self.return_results_as_name] = results
             results = scope
 
+        try:
+            validate_variable_names(results.keys())
```

Comment:
instead of doing 'per action plugin' (which user can override) do in results processing when the vars actually get created

Reply:
Do you mean https://github.com/ansible/ansible/blob/fbf2d5d/lib/ansible/plugins/strategy/__init__.py#L567? Can't that method also be overridden (unsure if someone would do that anyway)? It might be safer to put the validation to `VariableManager.set_host_variable()`. That method seems to be used only in the results processing for `include_vars` but it might not in the future? So I can't put `this invalid varname happened during include_vars` (which is useful information for the user) there because it might not be true when the method is used elsewhere.

Reply:
`set_host_variable` seems like the right place